sdk/eventhub/azure-eventhub/azure/eventhub/_producer_client.py

```diff
@@ -311,7 +312,9 @@ def send_batch(self, event_data_batch, **kwargs):
             cast(EventHubProducer, self._producers[partition_id]).send(
                 to_send_batch, timeout=send_timeout
             )
-        except (KeyError, AttributeError, EventHubError):
+        except (KeyError, AttributeError, EventHubError) as e:
+            _LOGGER.info(
+                "Producer for partition ID '{}' not available: {}. Rebuilding new producer.".format(partition_id, e))
```

Comment:
would this exception populate anything related to the token -- should it be debug level (pylint checker issue related)?

Reply:
The idea for this one is to figure out why are we going towards `start_producer` in the first place ( keyerror, object is None, or some other underlying issue that comes in ). 

Its after this point when the management request happens and there if/elif/else block that Ive added in that captures if the token is empty and if not print out the expires_on time.